servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -438,3 +454,146 @@ def test_query_jobs_submission_pagination(self):
+            job_failed = self.api_job_id(
+                self.start_job('echo SUCCEEDED', labels=labels, name=name))
+            self.wait_status(job_failed, ApiStatus.SUCCEEDED)
+
+            # Created a running job
+            job_running = self.api_job_id(
+                self.start_job('sleep 30', labels=labels, name=name))
+            self.wait_status(job_running, ApiStatus.RUNNING)
+
+            # Create a running job that has same label but no filter name
+            job_running = self.api_job_id(
+                self.start_job('sleep 30', labels=labels))
+            self.wait_status(job_running, ApiStatus.RUNNING)
+
+            # Create a running job that has different label and no filter name
+            job_running = self.api_job_id(
+                self.start_job('sleep 30', labels=labels_no_show))
+            self.wait_status(job_running, ApiStatus.RUNNING)
+
+            # Create a running job that has different label values
+            job_running = self.api_job_id(
+                self.start_job(
+                    'sleep 30', labels=other_values_labels, name=name))
+            self.wait_status(job_running, ApiStatus.RUNNING)
+
+            # Create a running job that has different labels
+            job_running = self.api_job_id(
+                self.start_job('sleep 30', labels=different_labels, name=name))
+            self.wait_status(job_running, ApiStatus.RUNNING)
+
+            status_counts_total = StatusCounts.from_dict({
+                'counts': [{
+                    'status': u'Running',
+                    'count': 3
+                }, {
+                    'status': u'Failed',
+                    'count': 1
+                }, {
+                    'status': u'Succeeded',
+                    'count': 1
+                }, {
+                    'status': u'Aborted',
+                    'count': 1
+                }]
+            })
+
+            status_counts_group = StatusCounts.from_dict({
+                'counts': [{
+                    'status': u'Running',
+                    'count': 1
+                }, {
+                    'status': u'Failed',
+                    'count': 1
+                }, {
+                    'status': u'Succeeded',
+                    'count': 1
+                }, {
+                    'status': u'Aborted',
+                    'count': 1
+                }]
+            })
+
+            status_counts_single = StatusCounts.from_dict({
+                'counts': [{
+                    'status': u'Running',
+                    'count': 1
+                }]
+            })
+
+            aggregation_resp = self.must_get_job_aggregations('HOURS_1')
+            self.assert_status_counts_equal(aggregation_resp.summary,
+                                            status_counts_total)
+
+            for aggregation in aggregation_resp.aggregations:
+                if aggregation.key == 'name' or aggregation.key == 'userId':
```

Comment:
In general it is clearer to avoid so much dynamic code in testing. Specifically just generating a single expected aggregations response above would make things much more readable IMO.

You can still make local helpers like `status_counts_single`.

```
status_counts_single =...
status_counts_group = ...
expected_response = AggregationResponse.from_dict({
  'counts': [
    {status: 'Running', count: 3},
    ...
  ],
  aggregations: [{
     key: 'name',
     entries: {
       label: name,
       status_counts: status_counts_total
     },
  }, {
    ...
  }]
})

assert_aggregation_response_equal(aggregation_resp, expected_response)
```